truelearn/learning/_knowledge_classifier.py

```diff
@@ -1,303 +1,161 @@
-
-    def fit(self, x: AbstractKnowledge, y: bool) -> KnowledgeClassifier:
-        """Train the model based on a given AbstractKnowledge that represents a learnable unit.
-
-        Parameters
-        ----------
-        x : AbstractKnowledge
-            A knowledge representation of a learnable unit.
-        y : bool
-            Whether the learner engages with the learnable unit.
-
-        Returns
-        -------
-        KnowledgeClassifier
-            The updated KnowledgeClassifier.
-
-        References
-        ----------
-        [1] Bulathwela, S. et al. (2020) “TrueLearn: A Family of Bayesian algorithms to match lifelong learners
-        to open educational resources,” Proceedings of the AAAI Conference on Artificial Intelligence, 34(01),
-        pp. 565-573. Available at: https://doi.org/10.1609/aaai.v34i01.5395.
-
-        """
-        if self.__positive_only and x is False:
-            return self
+        self._validate_params(
+            learner_model=learner_model,
+            threshold=threshold,
+            init_skill=init_skill,
+            def_var=def_var,
+            tau=tau,
+            beta=beta,
+            positive_only=positive_only,
+        )
 
-        learner_topic_kc_pairs = list(self.__select_topic_kc_pairs(x))
-        content_kcs = x.knowledge_components()
+        # the knowledge classifier doesn't rely on the draw probability
+        # it utilizes different assumptions
+        # so, we set draw probability to a very small value to avoid its impact
+        super().__init__(
+            learner_model=learner_model,
+            threshold=threshold,
+            init_skill=init_skill,
+            def_var=def_var,
+            tau=tau,
+            beta=beta,
+            positive_only=positive_only,
+            draw_proba_type="static",
+            draw_proba_static=KnowledgeClassifier.DRAW_PROBA_STATIC,
+            draw_proba_factor=0.1,
+        )
 
-        team_learner = tuple(
-            map(
-                lambda topic_kc_learner_pair: self.__env.create_rating(
-                    mu=topic_kc_learner_pair[1].mean, sigma=math.sqrt(topic_kc_learner_pair[1].variance)),
-                learner_topic_kc_pairs
+    def _update_knowledge_representation(self, x: EventModel, y: bool) -> None:
+        # make it a list because we need to use it more than one time later
+        learner_topic_kc_pairs = list(
+            select_topic_kc_pairs(
+                self.learner_model,
+                x.knowledge,
+                self.init_skill,
+                self.def_var,
             )
         )
-        team_content = tuple(
-            map(
-                lambda content_kc: self.__env.create_rating(
-                    mu=content_kc.mean, sigma=math.sqrt(content_kc.variance)),
-                content_kcs
-            )
+        learner_kcs = map(
+            lambda topic_kc_pair: topic_kc_pair[1], learner_topic_kc_pairs
```

Comment:
Shadows name 'topic_kc_pair' from outer scope 
Like i said in the other file: rename to learner_topic_kc_pair